benches/bench_slice.rs

```diff
@@ -0,0 +1,30 @@
+#![feature(test)]
```

Comment:
this doesn't really benchmark anything anymore, since the slice and Vec impls now just call out to the existing iterator impls